### PR TITLE
[RLlib] Fix flakiness supported action spaces tests

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/test_utils.py [hunk 1]
```diff
@@ -1481,7 +1481,7 @@ def check_supported_spaces(
     ]
 
     # TODO(Artur): Add back tf2 once we CNNs there
-    rlmodule_supported_frameworks = {"torch"}
+    rlmodule_supported_frameworks = "torch"
```
Comment (avnishn) [inline rllib/utils/test_utils.py:1484]: doesn't this need to be a list?

**Review by avnishn — COMMENTED** [review]
One question

File: rllib/utils/test_utils.py (hunk 1 above)
Comment (sven1977) [reply rllib/utils/test_utils.py:1484]: Also, we do now support CNNs in tf2, so we should resolve this TODO.

File: rllib/utils/test_utils.py [hunk 2]
```diff
@@ -1401,7 +1401,7 @@ def check_supported_spaces(
     config: "AlgorithmConfig",
     train: bool = True,
     check_bounds: bool = False,
-    frameworks: set = None,
+    frameworks: Optional[Sequence[str]] = None,
```
Comment (sven1977) [inline rllib/utils/test_utils.py:1404]: Can we please stop using `Sequence` (or `Mapping` for that matter)? A `Sequence` could also be a str. Instead, let's be more clear as to what we expect from the user. List is fine here.

Unrelated to this line: `Mapping` is also not great (I see this all over the place these days in RLlib). A Mapping per-se doesn't have the `update()` API for example, so IDEs consider these warnings and mark the code yellow if a method uses `.update` (or `get` or any other dict-specific API) on a var that was declared Mapping.

Comment (kouroshHakha) [reply rllib/utils/test_utils.py:1404]: Fair. Sequence is what was used for framework_iterator(). I'll change that here.

**Review by sven1977 — APPROVED** [review]
LGTM now. Thanks so much for the fixes @kouroshHakha !

**Pull request merged**